scripts/illumina_run_parameter_parser.py

```diff
@@ -0,0 +1,233 @@
+from datetime import datetime
+from genologics.lims import Lims
+from genologics.entities import Process
+from genologics.config import BASEURI, USERNAME, PASSWORD
+from flowcell_parser.classes import RunParser, RunParametersParser
+
+
+DESC = """EPP for parsing run paramters for Illumina MiSeq, NextSeq and NovaSeq runs"""
+
+
+def fetch_fc(process):
+    fc_id = ''
+    if process.parent_processes()[0].type.name == 'Load to Flowcell (NextSeq v1.0)':
+        if process.parent_processes()[0].udf['Experiment Name']:
+            fc_id = process.parent_processes()[0].udf['Experiment Name'].upper()
+        else:
+            sys.stderr.write("Experiment Name is empty in the associated Load to Flowcell (NextSeq v1.0) step.")
+            sys.exit(2)
+    elif process.parent_processes()[0].type.name == 'Denature, Dilute and Load Sample (MiSeq) 4.0':
+        if process.parent_processes()[0].udf['Flowcell ID']:
+            fc_id = process.parent_processes()[0].udf['Flowcell ID'].upper()
+        else:
+            sys.stderr.write("Flowcell ID is empty in the associated Denature, Dilute and Load Sample (MiSeq) 4.0 step.")
+            sys.exit(2)
+    elif process.parent_processes()[0].type.name == 'Load to Flowcell (NovaSeq 6000 v2.0)':
+        fc_id = process.parent_processes()[0].output_containers()[0].name
+    else:
+        sys.stderr.write("No associated parent step can be found.")
+        sys.exit(2)
+    return fc_id
+
+
+def fetch_rundir(fc_id, run_type):
+    run_dir = ''
+    if run_type == 'nextseq':
+        data_dir = 'NextSeq_data'
+    elif run_type == 'miseq':
+        data_dir = 'miseq_data'
+    elif run_type == 'novaseq':
+        data_dir = 'NovaSeq_data'
+    run_dir_path = os.path.join(os.sep,"srv","mfs",data_dir,"*{}".format(fc_id))
+    if len(glob.glob(run_dir_path)) == 1:
+        run_dir = glob.glob(run_dir_path)[0]
+    elif len(glob.glob(run_dir_path)) == 0:
+        sys.stderr.write("No run dir can be found for FC {}".format(fc_id))
+        sys.exit(2)
+    else:
+        sys.stderr.write("Multiple run dirs found for FC {}".format(fc_id))
+        sys.exit(2)
+    return run_dir
+
+
+def parse_run(run_dir):
+    runParserObj = RunParser(run_dir)
+    if os.path.exists("{}/RunParameters.xml".format(run_dir)):
+        RunParametersParserObj = RunParametersParser("{}/RunParameters.xml".format(run_dir))
+    elif os.path.exists("{}/runParameters.xml".format(run_dir)):
+        RunParametersParserObj = RunParametersParser("{}/runParameters.xml".format(run_dir))
+    else:
+        sys.stderr.write("No RunParameters.xml found for FC {}".format(fc_id))
+        sys.exit(2)
+    return runParserObj, RunParametersParserObj
+
+
+def attach_xml(process, run_dir):
+    for outart in process.all_outputs():
+        if outart.type == 'ResultFile' and outart.name == 'Run Info':
+            try:
+                lims.upload_new_file(outart, "{}/RunInfo.xml".format(run_dir))
+            except:
```

Comment:
Would be nice to have only the exact Exception here, so that it doesn't catch anything else

Reply:
Fixed